### ISSUE TITLE
Prev function doesn't work as expected

### ISSUE THREAD

**Issue by saikun0293** [richardscarrott/react-snap-carousel#1]
OS: Windows 10
Framework : Next.js
Issue : The prev function returned doesn't scroll to the prev page of elements

**Way to reproduce the error as an example.**
- There are 12 elements within the scroll container indexed from 0 to 11.
- The `pages` variable which holds indexes of elements which belong to a particular page sometimes ends up as follows =>`pages = [.....,[9,10],[11]]` . Here the last element alone belongs to the last page.
- In the UI however, the 9, 10, 11th elements appear on the last page. 
- Now clicking on the prev button it should target the leading element index i.e. 9 of the prev page and scroll there, but in the UI this 9th element is present in the last page itself hence scrollTo function doesn't scroll.
- It appears as if the prev function which is supposed to take me to `activePageIndex - 1` doesn't work

**Resolved by (not exactly the solution)**
I've resolved this by adding extra gap between the elements which i guess changes the allocation of elements to pages. Now the prev button works as expected.

**Comment by richardscarrott** [comment]
Hi @saikun0293, thanks the detailed report.

I've not been able to reproduce, but sounds like React Snap Carousel is measuring the DOM incorrectly; perhaps a rounding error if you're using percentage widths?

It'd be really helpful if you could fork this [CodeSandbox](https://codesandbox.io/s/react-snap-carousel-0zlvmw?file=/src/Carousel.tsx) demonstrating the issue so I can investigate & push a fix?

Failing that, could you paste the entire `pages` array as then I might be able to reproduce myself.

**Comment by saikun0293** [comment]
`pages = [
    [
        0,
        1,
        2
    ],
    [
        3,
        4
    ],
    [
        5,
        6
    ],
    [
        7,
        8
    ],
    [
        9,
        10
    ],
    [
        11
    ]
]`

![image](https://user-images.githubusercontent.com/56252312/212021135-e29d5140-c420-4111-94b3-67b2c478ed90.png)

I've tried in codesandbox but it works there as intended.

Widths
`scroll-container = 1215.35px`
`scroll-item = 384px`
`gap = 24px` between scroll-items

```css
/* Carousel styles */
.scroll-container {
  position: relative;
  display: flex;
  overflow: auto;
  scroll-snap-type: x mandatory;

  -ms-overflow-style: none; /* Internet Explorer 10+ */
  scrollbar-width: none; /* Firefox */
}
.scroll-container::-webkit-scrollbar {
  display: none; /* Safari and Chrome */
}
.scroll-item {
  flex-shrink: 0;
}

```
Hope this helps

**Comment by richardscarrott** [comment]
Thanks, I've managed to reproduce given the above.

I have a fix which I'll push up either today or tomorrow 👍

**Comment by richardscarrott** [comment]
@saikun0293 I've published a new beta version which you can test out using `npm add react-snap-carousel@0.1.0-next.0`.

Please give it a go, and if it works okay for you let me know and I'll publish as `0.1.0`.

**Comment by richardscarrott** [comment]
For my future reference, this codesandbox reproduces the bug https://codesandbox.io/s/react-snap-carousel-1-bug-5uu40e?file=/src/App.tsx:92-104

**Comment by saikun0293** [comment]
Yup, it works 🚀 tested it out @richardscarrott

**Comment by richardscarrott** [comment]
Great, I've published under the `latest` tag as 0.1.0 https://www.npmjs.com/package/react-snap-carousel/v/0.1.0 so you can install with `npm add react-snap-carousel`.
